{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Tlss3vMDrZm_"
      },
      "source": [
        "# Домашнее задание 1 по анализу bulk RNA-Seq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "xuH2wHLQrfnC"
      },
      "source": [
        "## Часть 1. Контроль качества и картирование"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "VgGlbg4NrjXH"
      },
      "source": [
        "Для начала установим необходимые нам инструменты: `FastQC` и `kallisto`."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bRkHovNSqr_1",
        "outputId": "fe0e56f2-ea64-4c0f-fa52-03a0ff5674c4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'kallisto'...\n",
            "remote: Enumerating objects: 8080, done.\u001b[K\n",
            "remote: Counting objects: 100% (1162/1162), done.\u001b[K\n",
            "remote: Compressing objects: 100% (314/314), done.\u001b[K\n",
            "remote: Total 8080 (delta 880), reused 984 (delta 846), pack-reused 6918\u001b[K\n",
            "Receiving objects: 100% (8080/8080), 9.11 MiB | 11.00 MiB/s, done.\n",
            "Resolving deltas: 100% (5306/5306), done.\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "autoconf is already the newest version (2.71-2).\n",
            "autoconf set to manually installed.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 9 not upgraded.\n",
            "\u001b[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\u001b[0m\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17 - Success\n",
            "\u001b[0mshared build\u001b[0m\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE  \n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")  \n",
            "-- Configuring done (1.6s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/build\n",
            "[  3%] \u001b[34m\u001b[1mCreating directories for 'bifrost'\u001b[0m\n",
            "[  7%] \u001b[34m\u001b[1mNo download step for 'bifrost'\u001b[0m\n",
            "[ 11%] \u001b[34m\u001b[1mNo update step for 'bifrost'\u001b[0m\n",
            "[ 15%] \u001b[34m\u001b[1mNo patch step for 'bifrost'\u001b[0m\n",
            "[ 19%] \u001b[34m\u001b[1mPerforming configure step for 'bifrost'\u001b[0m\n",
            "\u001b[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\u001b[0m\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE  \n",
            "\u001b[0mCompilation architecture: native\u001b[0m\n",
            "\u001b[0mBuild type: Release\u001b[0m\n",
            "\u001b[0mMaximum k-mer size: 31\u001b[0m\n",
            "\u001b[0mMaximum g-mer size: 31\u001b[0m\n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")  \n",
            "-- Configuring done (0.8s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/ext/bifrost/build\n",
            "[ 23%] \u001b[34m\u001b[1mPerforming build step for 'bifrost'\u001b[0m\n",
            "[  3%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/Bifrost.cpp.o\u001b[0m\n",
            "[  6%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/BitContainer.cpp.o\u001b[0m\n",
            "[  9%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/BlockedBloomFilter.cpp.o\u001b[0m\n",
            "[ 12%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/ColorSet.cpp.o\u001b[0m\n",
            "[ 15%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/CompressedCoverage.cpp.o\u001b[0m\n",
            "[ 18%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/CompressedSequence.cpp.o\u001b[0m\n",
            "[ 21%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/FASTX_Parser.cpp.o\u001b[0m\n",
            "[ 25%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/GFA_Parser.cpp.o\u001b[0m\n",
            "[ 28%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/Kmer.cpp.o\u001b[0m\n",
            "[ 31%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/KmerIterator.cpp.o\u001b[0m\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/MinimizerIndex.cpp.o\u001b[0m\n",
            "[ 37%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/TinyBitmap.cpp.o\u001b[0m\n",
            "[ 40%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/UnitigMap.cpp.o\u001b[0m\n",
            "[ 43%] \u001b[32mBuilding C object src/CMakeFiles/bifrost_static.dir/roaring.c.o\u001b[0m\n",
            "[ 46%] \u001b[32m\u001b[1mLinking CXX static library libbifrost.a\u001b[0m\n",
            "[ 46%] Built target bifrost_static\n",
            "[ 50%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/Bifrost.cpp.o\u001b[0m\n",
            "[ 53%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/BitContainer.cpp.o\u001b[0m\n",
            "[ 56%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/BlockedBloomFilter.cpp.o\u001b[0m\n",
            "[ 59%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/ColorSet.cpp.o\u001b[0m\n",
            "[ 62%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedCoverage.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedSequence.cpp.o\u001b[0m\n",
            "[ 68%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/FASTX_Parser.cpp.o\u001b[0m\n",
            "[ 71%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/GFA_Parser.cpp.o\u001b[0m\n",
            "[ 75%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/Kmer.cpp.o\u001b[0m\n",
            "[ 78%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/KmerIterator.cpp.o\u001b[0m\n",
            "[ 81%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/MinimizerIndex.cpp.o\u001b[0m\n",
            "[ 84%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/TinyBitmap.cpp.o\u001b[0m\n",
            "[ 87%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/UnitigMap.cpp.o\u001b[0m\n",
            "[ 90%] \u001b[32mBuilding C object src/CMakeFiles/bifrost_dynamic.dir/roaring.c.o\u001b[0m\n",
            "[ 93%] \u001b[32m\u001b[1mLinking CXX shared library libbifrost.so\u001b[0m\n",
            "[ 93%] Built target bifrost_dynamic\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/Bifrost.dir/Bifrost.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable Bifrost\u001b[0m\n",
            "[100%] Built target Bifrost\n",
            "[ 26%] \u001b[34m\u001b[1mNo install step for 'bifrost'\u001b[0m\n",
            "[ 30%] \u001b[34m\u001b[1mCompleted 'bifrost'\u001b[0m\n",
            "[ 30%] Built target bifrost\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSData.cpp.o\u001b[0m\n",
            "[ 38%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSTools.cpp.o\u001b[0m\n",
            "[ 42%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Bootstrap.cpp.o\u001b[0m\n",
            "[ 46%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/GeneModel.cpp.o\u001b[0m\n",
            "[ 50%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/H5Writer.cpp.o\u001b[0m\n",
            "[ 53%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIndex.cpp.o\u001b[0m\n",
            "[ 57%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/MinCollector.cpp.o\u001b[0m\n",
            "[ 61%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PlaintextWriter.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/ProcessReads.cpp.o\u001b[0m\n",
            "[ 69%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PseudoBam.cpp.o\u001b[0m\n",
            "[ 73%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/common.cpp.o\u001b[0m\n",
            "[ 76%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/h5utils.cpp.o\u001b[0m\n",
            "[ 80%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/hash.cpp.o\u001b[0m\n",
            "[ 84%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/main.cpp.o\u001b[0m\n",
            "[ 88%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/weights.cpp.o\u001b[0m\n",
            "[ 92%] \u001b[32m\u001b[1mLinking CXX static library libkallisto_core.a\u001b[0m\n",
            "[ 92%] Built target kallisto_core\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto.dir/main.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable kallisto\u001b[0m\n",
            "[100%] Built target kallisto\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  default-jre default-jre-headless fonts-dejavu-core fonts-dejavu-extra\n",
            "  libapache-pom-java libargs4j-java libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libcommons-compress-java libcommons-io-java\n",
            "  libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl\n",
            "  libfontenc1 libhtsjdk-java libjbzip2-java libjson-simple-java libngs-java\n",
            "  libngs-sdk-dev libngs-sdk2 libsis-base-java libsis-base-jni\n",
            "  libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "Suggested packages:\n",
            "  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc\n",
            "  libavalon-framework-java libcommons-logging-java-doc\n",
            "  libexcalibur-logkit-java liblog4j1.2-java picard-tools libjson-simple-doc\n",
            "  mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  default-jre default-jre-headless fastqc fonts-dejavu-core fonts-dejavu-extra\n",
            "  libapache-pom-java libargs4j-java libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libcommons-compress-java libcommons-io-java\n",
            "  libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl\n",
            "  libfontenc1 libhtsjdk-java libjbzip2-java libjson-simple-java libngs-java\n",
            "  libngs-sdk-dev libngs-sdk2 libsis-base-java libsis-base-jni\n",
            "  libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "0 upgraded, 35 newly installed, 0 to remove and 9 not upgraded.\n",
            "Need to get 11.8 MB of archives.\n",
            "After this operation, 24.8 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.20.1+1-0ubuntu1~22.04 [213 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Fetched 11.8 MB in 1s (23.2 MB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 35.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package default-jre-headless.\n",
            "(Reading database ... 120880 files and directories currently installed.)\n",
            "Preparing to unpack .../00-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre-headless (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../01-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package openjdk-11-jre:amd64.\n",
            "Preparing to unpack .../02-openjdk-11-jre_11.0.20.1+1-0ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-11-jre:amd64 (11.0.20.1+1-0ubuntu1~22.04) ...\n",
            "Selecting previously unselected package default-jre.\n",
            "Preparing to unpack .../03-default-jre_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "Preparing to unpack .../04-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../05-libjbzip2-java_0.9.1-7_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-7) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../06-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../07-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../08-libcommons-compress-java_1.21-1_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.21-1) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../09-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../10-libcommons-jexl2-java_2.1.1-5_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-5) ...\n",
            "Selecting previously unselected package libjson-simple-java.\n",
            "Preparing to unpack .../11-libjson-simple-java_2.3.0-1_all.deb ...\n",
            "Unpacking libjson-simple-java (2.3.0-1) ...\n",
            "Selecting previously unselected package libngs-sdk2:amd64.\n",
            "Preparing to unpack .../12-libngs-sdk2_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev:amd64.\n",
            "Preparing to unpack .../13-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-java:amd64.\n",
            "Preparing to unpack .../14-libngs-java_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-java:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../15-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../16-libsnappy-java_1.1.8.3-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../17-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Selecting previously unselected package libfindbin-libs-perl.\n",
            "Preparing to unpack .../18-libfindbin-libs-perl_2.190.02-1_all.deb ...\n",
            "Unpacking libfindbin-libs-perl (2.190.02-1) ...\n",
            "Selecting previously unselected package libargs4j-java.\n",
            "Preparing to unpack .../19-libargs4j-java_2.33-1.1_all.deb ...\n",
            "Unpacking libargs4j-java (2.33-1.1) ...\n",
            "Selecting previously unselected package libcommons-io-java.\n",
            "Preparing to unpack .../20-libcommons-io-java_2.11.0-2_all.deb ...\n",
            "Unpacking libcommons-io-java (2.11.0-2) ...\n",
            "Selecting previously unselected package libcommons-lang3-java.\n",
            "Preparing to unpack .../21-libcommons-lang3-java_3.11-1_all.deb ...\n",
            "Unpacking libcommons-lang3-java (3.11-1) ...\n",
            "Selecting previously unselected package libsis-base-jni.\n",
            "Preparing to unpack .../22-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n",
            "Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-base-java.\n",
            "Preparing to unpack .../23-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n",
            "Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-jhdf5-jni.\n",
            "Preparing to unpack .../24-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n",
            "Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package libsis-jhdf5-java.\n",
            "Preparing to unpack .../25-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n",
            "Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../26-fastqc_0.11.9+dfsg-5_all.deb ...\n",
            "Unpacking fastqc (0.11.9+dfsg-5) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../27-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../28-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libfontenc1:amd64.\n",
            "Preparing to unpack .../29-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n",
            "Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Selecting previously unselected package libxkbfile1:amd64.\n",
            "Preparing to unpack .../30-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n",
            "Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../31-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../32-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../33-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../34-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up default-jre-headless (2:1.11-72build2) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up openjdk-11-jre:amd64 (11.0.20.1+1-0ubuntu1~22.04) ...\n",
            "Setting up libfindbin-libs-perl (2.190.02-1) ...\n",
            "Setting up default-jre (2:1.11-72build2) ...\n",
            "Setting up libjson-simple-java (2.3.0-1) ...\n",
            "Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libjbzip2-java (0.9.1-7) ...\n",
            "Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Setting up libngs-java:amd64 (2.11.2-3) ...\n",
            "Setting up libsnappy-jni (1.1.8.3-1) ...\n",
            "Setting up libargs4j-java (2.33-1.1) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-5) ...\n",
            "Setting up libcommons-lang3-java (3.11-1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libsnappy-java (1.1.8.3-1) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libcommons-compress-java (1.21-1) ...\n",
            "Setting up libcommons-io-java (2.11.0-2) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Setting up fastqc (0.11.9+dfsg-5) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# Установка kallisto\n",
        "!git clone https://github.com/pachterlab/kallisto.git\n",
        "!apt-get install autoconf\n",
        "!cd kallisto && mkdir build && cd build && cmake .. && make\n",
        "\n",
        "# Установка FastQC\n",
        "!sudo apt-get install fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "M-4NLFhFrrq6"
      },
      "source": [
        "Теперь загрузим прочтения эксперимента bulk RNA-Seq (для получения прямых ссылок был использован ресурс [SRA-Explorer](https://sra-explorer.info))."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EWfc1a7fa97F",
        "outputId": "8d0fb226-f6fd-4fb8-cb63-fdfa24e882bb"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1638M  100 1638M    0     0  44.2M      0  0:00:37  0:00:37 --:--:-- 45.9M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1646M  100 1646M    0     0  38.1M      0  0:00:43  0:00:43 --:--:-- 41.3M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1635M  100 1635M    0     0  27.8M      0  0:00:58  0:00:58 --:--:-- 36.3M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1643M  100 1643M    0     0  35.9M      0  0:00:45  0:00:45 --:--:-- 40.8M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1634M  100 1634M    0     0  39.2M      0  0:00:41  0:00:41 --:--:-- 41.6M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1643M  100 1643M    0     0  38.9M      0  0:00:42  0:00:42 --:--:-- 41.8M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1624M  100 1624M    0     0  39.4M      0  0:00:41  0:00:41 --:--:-- 41.7M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1632M  100 1632M    0     0  39.1M      0  0:00:41  0:00:41 --:--:-- 40.8M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1690M  100 1690M    0     0  39.3M      0  0:00:43  0:00:43 --:--:-- 41.9M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1692M  100 1692M    0     0  38.9M      0  0:00:43  0:00:43 --:--:-- 40.7M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1624M  100 1624M    0     0  39.0M      0  0:00:41  0:00:41 --:--:-- 41.2M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1634M  100 1634M    0     0  39.2M      0  0:00:41  0:00:41 --:--:-- 41.1M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1640M  100 1640M    0     0  39.2M      0  0:00:41  0:00:41 --:--:-- 39.8M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1647M  100 1647M    0     0  39.0M      0  0:00:42  0:00:42 --:--:-- 41.5M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1628M  100 1628M    0     0  38.8M      0  0:00:41  0:00:41 --:--:-- 40.2M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1638M  100 1638M    0     0  39.1M      0  0:00:41  0:00:41 --:--:-- 39.8M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1618M  100 1618M    0     0  39.3M      0  0:00:41  0:00:41 --:--:-- 41.7M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1627M  100 1627M    0     0  38.7M      0  0:00:41  0:00:41 --:--:-- 40.9M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1687M  100 1687M    0     0  39.4M      0  0:00:42  0:00:42 --:--:-- 41.7M\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 1691M  100 1691M    0     0  39.4M      0  0:00:42  0:00:42 --:--:-- 41.6M\n"
          ]
        }
      ],
      "source": [
        "#B1\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/003/SRR6410613/SRR6410613_1.fastq.gz \\\n",
        "  -o B1_SRR6410613_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/003/SRR6410613/SRR6410613_2.fastq.gz \\\n",
        "  -o B1_SRR6410613_2.fastq.gz\n",
        "\n",
        "#B2\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/004/SRR6410614/SRR6410614_1.fastq.gz \\\n",
        "  -o B2_SRR6410614_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/004/SRR6410614/SRR6410614_2.fastq.gz \\\n",
        "  -o B2_SRR6410614_2.fastq.gz\n",
        "\n",
        "#B3\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/001/SRR6410611/SRR6410611_1.fastq.gz \\\n",
        "  -o B3_SRR6410611_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/001/SRR6410611/SRR6410611_2.fastq.gz \\\n",
        "  -o B3_SRR6410611_2.fastq.gz\n",
        "\n",
        "#B4\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/002/SRR6410612/SRR6410612_1.fastq.gz \\\n",
        "  -o B4_SRR6410612_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/002/SRR6410612/SRR6410612_2.fastq.gz \\\n",
        "  -o B4_SRR6410612_2.fastq.gz\n",
        "\n",
        "#B5\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/005/SRR6410615/SRR6410615_1.fastq.gz \\\n",
        "  -o B5_SRR6410615_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/005/SRR6410615/SRR6410615_2.fastq.gz \\\n",
        "  -o B5_SRR6410615_2.fastq.gz\n",
        "\n",
        "#A1\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/005/SRR6410605/SRR6410605_1.fastq.gz \\\n",
        "  -o A1_SRR6410605_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/005/SRR6410605/SRR6410605_2.fastq.gz \\\n",
        "  -o A1_SRR6410605_2.fastq.gz\n",
        "\n",
        "#A2\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/006/SRR6410606/SRR6410606_1.fastq.gz \\\n",
        "  -o A2_SRR6410606_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/006/SRR6410606/SRR6410606_2.fastq.gz \\\n",
        "  -o A2_SRR6410606_2.fastq.gz\n",
        "\n",
        "#A3\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/003/SRR6410603/SRR6410603_1.fastq.gz \\\n",
        "  -o A3_SRR6410603_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/003/SRR6410603/SRR6410603_2.fastq.gz \\\n",
        "  -o A3_SRR6410603_2.fastq.gz\n",
        "\n",
        "#A4\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/004/SRR6410604/SRR6410604_1.fastq.gz \\\n",
        "  -o A4_SRR6410604_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/004/SRR6410604/SRR6410604_2.fastq.gz \\\n",
        "  -o A4_SRR6410604_2.fastq.gz\n",
        "\n",
        "#A5\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/009/SRR6410609/SRR6410609_1.fastq.gz \\\n",
        "  -o A5_SRR6410609_1.fastq.gz\n",
        "\n",
        "!curl -L \\\n",
        "  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR641/009/SRR6410609/SRR6410609_2.fastq.gz \\\n",
        "  -o A5_SRR6410609_2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ob5FMAd8sbRx"
      },
      "source": [
        "Оценим качество прочтений при помощи `FastQC`:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EkB61VIqsORv",
        "outputId": "08ff4d3d-4953-4646-d889-d22c5ab09675"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "mkdir: cannot create directory ‘/content/fastqc_results’: File exists\n",
            "Started analysis of B1_SRR6410613_1.fastq.gz\n",
            "Started analysis of B1_SRR6410613_2.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "!mkdir /content/fastqc_results\n",
        "!fastqc -o /content/fastqc_results B1_SRR6410613_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B1_SRR6410613_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5nQlTfaylFkM",
        "outputId": "e5795e33-5498-4847-a9b9-485623bd266e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Started analysis of B2_SRR6410614_1.fastq.gz\n",
            "Approx 5% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 10% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 15% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 20% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 25% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 30% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 35% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 40% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 45% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 50% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 55% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 60% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 65% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 70% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 75% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 80% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 85% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 90% complete for B2_SRR6410614_1.fastq.gz\n",
            "Approx 95% complete for B2_SRR6410614_1.fastq.gz\n",
            "Analysis complete for B2_SRR6410614_1.fastq.gz\n",
            "Started analysis of B2_SRR6410614_2.fastq.gz\n",
            "Approx 5% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 10% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 15% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 20% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 25% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 30% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 35% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 40% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 45% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 50% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 55% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 60% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 65% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 70% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 75% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 80% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 85% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 90% complete for B2_SRR6410614_2.fastq.gz\n",
            "Approx 95% complete for B2_SRR6410614_2.fastq.gz\n",
            "Analysis complete for B2_SRR6410614_2.fastq.gz\n",
            "Started analysis of B3_SRR6410611_1.fastq.gz\n",
            "Approx 5% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 10% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 15% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 20% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 25% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 30% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 35% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 40% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 45% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 50% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 55% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 60% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 65% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 70% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 75% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 80% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 85% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 90% complete for B3_SRR6410611_1.fastq.gz\n",
            "Approx 95% complete for B3_SRR6410611_1.fastq.gz\n",
            "Analysis complete for B3_SRR6410611_1.fastq.gz\n",
            "Started analysis of B3_SRR6410611_2.fastq.gz\n",
            "Approx 5% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 10% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 15% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 20% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 25% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 30% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 35% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 40% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 45% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 50% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 55% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 60% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 65% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 70% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 75% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 80% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 85% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 90% complete for B3_SRR6410611_2.fastq.gz\n",
            "Approx 95% complete for B3_SRR6410611_2.fastq.gz\n",
            "Analysis complete for B3_SRR6410611_2.fastq.gz\n",
            "Started analysis of B4_SRR6410612_1.fastq.gz\n",
            "Approx 5% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 10% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 15% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 20% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 25% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 30% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 35% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 40% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 45% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 50% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 55% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 60% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 65% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 70% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 75% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 80% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 85% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 90% complete for B4_SRR6410612_1.fastq.gz\n",
            "Approx 95% complete for B4_SRR6410612_1.fastq.gz\n",
            "Analysis complete for B4_SRR6410612_1.fastq.gz\n",
            "Started analysis of B4_SRR6410612_2.fastq.gz\n",
            "Approx 5% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 10% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 15% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 20% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 25% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 30% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 35% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 40% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 45% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 50% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 55% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 60% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 65% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 70% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 75% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 80% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 85% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 90% complete for B4_SRR6410612_2.fastq.gz\n",
            "Approx 95% complete for B4_SRR6410612_2.fastq.gz\n",
            "Analysis complete for B4_SRR6410612_2.fastq.gz\n",
            "Started analysis of B5_SRR6410615_1.fastq.gz\n",
            "Approx 5% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 10% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 15% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 20% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 25% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 30% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 35% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 40% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 45% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 50% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 55% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 60% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 65% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 70% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 75% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 80% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 85% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 90% complete for B5_SRR6410615_1.fastq.gz\n",
            "Approx 95% complete for B5_SRR6410615_1.fastq.gz\n",
            "Analysis complete for B5_SRR6410615_1.fastq.gz\n",
            "Started analysis of B5_SRR6410615_2.fastq.gz\n",
            "Approx 5% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 10% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 15% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 20% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 25% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 30% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 35% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 40% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 45% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 50% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 55% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 60% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 65% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 70% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 75% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 80% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 85% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 90% complete for B5_SRR6410615_2.fastq.gz\n",
            "Approx 95% complete for B5_SRR6410615_2.fastq.gz\n",
            "Analysis complete for B5_SRR6410615_2.fastq.gz\n",
            "Started analysis of A1_SRR6410605_1.fastq.gz\n",
            "Approx 5% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 10% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 15% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 20% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 25% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 30% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 35% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 40% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 45% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 50% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 55% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 60% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 65% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 70% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 75% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 80% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 85% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 90% complete for A1_SRR6410605_1.fastq.gz\n",
            "Approx 95% complete for A1_SRR6410605_1.fastq.gz\n",
            "Analysis complete for A1_SRR6410605_1.fastq.gz\n",
            "Started analysis of A1_SRR6410605_2.fastq.gz\n",
            "Approx 5% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 10% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 15% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 20% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 25% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 30% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 35% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 40% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 45% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 50% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 55% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 60% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 65% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 70% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 75% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 80% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 85% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 90% complete for A1_SRR6410605_2.fastq.gz\n",
            "Approx 95% complete for A1_SRR6410605_2.fastq.gz\n",
            "Analysis complete for A1_SRR6410605_2.fastq.gz\n",
            "Started analysis of A2_SRR6410606_1.fastq.gz\n",
            "Approx 5% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 10% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 15% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 20% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 25% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 30% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 35% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 40% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 45% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 50% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 55% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 60% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 65% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 70% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 75% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 80% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 85% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 90% complete for A2_SRR6410606_1.fastq.gz\n",
            "Approx 95% complete for A2_SRR6410606_1.fastq.gz\n",
            "Analysis complete for A2_SRR6410606_1.fastq.gz\n",
            "Started analysis of A2_SRR6410606_2.fastq.gz\n",
            "Approx 5% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 10% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 15% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 20% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 25% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 30% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 35% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 40% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 45% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 50% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 55% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 60% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 65% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 70% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 75% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 80% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 85% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 90% complete for A2_SRR6410606_2.fastq.gz\n",
            "Approx 95% complete for A2_SRR6410606_2.fastq.gz\n",
            "Analysis complete for A2_SRR6410606_2.fastq.gz\n",
            "Started analysis of A3_SRR6410603_1.fastq.gz\n",
            "Approx 5% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 10% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 15% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 20% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 25% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 30% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 35% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 40% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 45% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 50% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 55% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 60% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 65% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 70% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 75% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 80% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 85% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 90% complete for A3_SRR6410603_1.fastq.gz\n",
            "Approx 95% complete for A3_SRR6410603_1.fastq.gz\n",
            "Analysis complete for A3_SRR6410603_1.fastq.gz\n",
            "Started analysis of A3_SRR6410603_2.fastq.gz\n",
            "Approx 5% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 10% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 15% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 20% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 25% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 30% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 35% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 40% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 45% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 50% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 55% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 60% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 65% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 70% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 75% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 80% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 85% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 90% complete for A3_SRR6410603_2.fastq.gz\n",
            "Approx 95% complete for A3_SRR6410603_2.fastq.gz\n",
            "Analysis complete for A3_SRR6410603_2.fastq.gz\n",
            "Started analysis of A4_SRR6410604_1.fastq.gz\n",
            "Approx 5% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 10% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 15% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 20% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 25% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 30% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 35% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 40% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 45% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 50% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 55% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 60% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 65% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 70% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 75% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 80% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 85% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 90% complete for A4_SRR6410604_1.fastq.gz\n",
            "Approx 95% complete for A4_SRR6410604_1.fastq.gz\n",
            "Analysis complete for A4_SRR6410604_1.fastq.gz\n",
            "Started analysis of A4_SRR6410604_2.fastq.gz\n",
            "Approx 5% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 10% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 15% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 20% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 25% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 30% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 35% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 40% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 45% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 50% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 55% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 60% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 65% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 70% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 75% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 80% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 85% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 90% complete for A4_SRR6410604_2.fastq.gz\n",
            "Approx 95% complete for A4_SRR6410604_2.fastq.gz\n",
            "Analysis complete for A4_SRR6410604_2.fastq.gz\n",
            "Started analysis of A5_SRR6410609_1.fastq.gz\n",
            "Approx 5% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 10% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 15% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 20% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 25% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 30% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 35% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 40% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 45% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 50% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 55% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 60% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 65% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 70% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 75% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 80% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 85% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 90% complete for A5_SRR6410609_1.fastq.gz\n",
            "Approx 95% complete for A5_SRR6410609_1.fastq.gz\n",
            "Analysis complete for A5_SRR6410609_1.fastq.gz\n",
            "Started analysis of A5_SRR6410609_2.fastq.gz\n",
            "Approx 5% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 10% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 15% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 20% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 25% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 30% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 35% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 40% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 45% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 50% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 55% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 60% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 65% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 70% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 75% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 80% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 85% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 90% complete for A5_SRR6410609_2.fastq.gz\n",
            "Approx 95% complete for A5_SRR6410609_2.fastq.gz\n",
            "Analysis complete for A5_SRR6410609_2.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "!fastqc -o /content/fastqc_results B2_SRR6410614_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B2_SRR6410614_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B3_SRR6410611_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B3_SRR6410611_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B4_SRR6410612_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B4_SRR6410612_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B5_SRR6410615_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results B5_SRR6410615_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A1_SRR6410605_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A1_SRR6410605_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A2_SRR6410606_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A2_SRR6410606_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A3_SRR6410603_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A3_SRR6410603_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A4_SRR6410604_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A4_SRR6410604_2.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A5_SRR6410609_1.fastq.gz\n",
        "!fastqc -o /content/fastqc_results A5_SRR6410609_2.fastq.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bj81klh0nplX",
        "outputId": "c82ca87e-d9dc-4dc2-c3ad-6d2ab756a335"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting multiqc\n",
            "  Downloading multiqc-1.18-py3-none-any.whl (1.4 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.4/1.4 MB\u001b[0m \u001b[31m10.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: matplotlib>=2.1.1 in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.7.1)\n",
            "Requirement already satisfied: networkx>=2.5.1 in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.2.1)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from multiqc) (1.23.5)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from multiqc) (8.1.7)\n",
            "Collecting coloredlogs (from multiqc)\n",
            "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m6.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: future>0.14.0 in /usr/local/lib/python3.10/dist-packages (from multiqc) (0.18.3)\n",
            "Requirement already satisfied: jinja2>=3.0.0 in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.1.2)\n",
            "Requirement already satisfied: markdown in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.5.1)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from multiqc) (23.2)\n",
            "Requirement already satisfied: pyyaml>=4 in /usr/local/lib/python3.10/dist-packages (from multiqc) (6.0.1)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from multiqc) (2.31.0)\n",
            "Requirement already satisfied: rich>=10 in /usr/local/lib/python3.10/dist-packages (from multiqc) (13.7.0)\n",
            "Collecting rich-click (from multiqc)\n",
            "  Downloading rich_click-1.7.1-py3-none-any.whl (32 kB)\n",
            "Collecting spectra>=0.0.10 (from multiqc)\n",
            "  Downloading spectra-0.0.11.tar.gz (18 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/lib/python3.10/dist-packages (from multiqc) (6.8.0)\n",
            "Requirement already satisfied: humanize in /usr/local/lib/python3.10/dist-packages (from multiqc) (4.7.0)\n",
            "Collecting pyaml-env (from multiqc)\n",
            "  Downloading pyaml_env-1.2.1-py3-none-any.whl (9.0 kB)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2>=3.0.0->multiqc) (2.1.3)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (1.2.0)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (4.44.3)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (1.4.5)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (9.4.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (3.1.1)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.10/dist-packages (from matplotlib>=2.1.1->multiqc) (2.8.2)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich>=10->multiqc) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich>=10->multiqc) (2.16.1)\n",
            "Collecting colormath>=3.0.0 (from spectra>=0.0.10->multiqc)\n",
            "  Downloading colormath-3.0.0.tar.gz (39 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting humanfriendly>=9.1 (from coloredlogs->multiqc)\n",
            "  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m13.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.10/dist-packages (from importlib-metadata->multiqc) (3.17.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (3.4)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (2023.7.22)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from rich-click->multiqc) (4.5.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich>=10->multiqc) (0.1.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.7->matplotlib>=2.1.1->multiqc) (1.16.0)\n",
            "Building wheels for collected packages: spectra, colormath\n",
            "  Building wheel for spectra (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for spectra: filename=spectra-0.0.11-py3-none-any.whl size=17467 sha256=8da73a1c157df40716e42914b4eb4b94eee5ca3f5043a2f83fbaefa4f62c591c\n",
            "  Stored in directory: /root/.cache/pip/wheels/bc/27/ec/cddaa9489a49ddcf702e1f4dc1791db28a3b0121e1ff254f97\n",
            "  Building wheel for colormath (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for colormath: filename=colormath-3.0.0-py3-none-any.whl size=39405 sha256=b17bcfc3c4833f3192bff5f36ab53b33ce786e877a1748e57eb11ebcb3cca86c\n",
            "  Stored in directory: /root/.cache/pip/wheels/ab/b3/4d/c0738759c25a1df01958068f162cf2a9dc3ab1da8b972cfcfc\n",
            "Successfully built spectra colormath\n",
            "Installing collected packages: pyaml-env, humanfriendly, colormath, spectra, coloredlogs, rich-click, multiqc\n",
            "Successfully installed coloredlogs-15.0.1 colormath-3.0.0 humanfriendly-10.0 multiqc-1.18 pyaml-env-1.2.1 rich-click-1.7.1 spectra-0.0.11\n"
          ]
        }
      ],
      "source": [
        "!pip install multiqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iHTu_HYmnxqi",
        "outputId": "294bafc0-94df-4964-b42b-68416a7d395e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "  \u001b[91m///\u001b[0m \u001b]8;id=253347;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.18\u001b[0m\n",
            "\n",
            "\u001b[34m|           multiqc\u001b[0m | Search path : /content/fastqc_results\n",
            "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m40/40\u001b[0m  \n",
            "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 20 reports\n",
            "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report.html\n",
            "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data\n",
            "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
          ]
        }
      ],
      "source": [
        "!multiqc /content/fastqc_results/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "t8u1TT-Itop3"
      },
      "source": [
        "Подготовка референса для `kallisto`:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_N_wqZWOsjAx",
        "outputId": "0cab3c93-848e-4f7b-f04d-0cba781615c8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 64.7M  100 64.7M    0     0  16.3M      0  0:00:03  0:00:03 --:--:-- 16.3M\n",
            "\n",
            "[build] loading fasta file Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "[build] k-mer length: 31\n",
            "[build] warning: clipped off poly-A tail (longer than 10)\n",
            "        from 1471 target sequences\n",
            "[build] warning: replaced 5 non-ACGUT characters in the input sequence\n",
            "        with pseudorandom nucleotides\n",
            "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
            "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
            "KmerStream::KmerStream(): Finished\n",
            "CompactedDBG::build(): Estimated number of k-mers occurring at least once: 108613141\n",
            "CompactedDBG::build(): Estimated number of minimizer occurring at least once: 26591029\n",
            "CompactedDBG::filter(): Processed 304594999 k-mers in 187626 reads\n",
            "CompactedDBG::filter(): Found 108523609 unique k-mers\n",
            "CompactedDBG::filter(): Number of blocks in Bloom filter is 742474\n",
            "CompactedDBG::construct(): Extract approximate unitigs (1/2)\n",
            "CompactedDBG::construct(): Extract approximate unitigs (2/2)\n",
            "CompactedDBG::construct(): Closed all input files\n",
            "\n",
            "CompactedDBG::construct(): Splitting unitigs (1/2)\n",
            "\n",
            "CompactedDBG::construct(): Splitting unitigs (2/2)\n",
            "CompactedDBG::construct(): Before split: 991986 unitigs\n",
            "CompactedDBG::construct(): After split (1/1): 991986 unitigs\n",
            "CompactedDBG::construct(): Unitigs split: 1716\n",
            "CompactedDBG::construct(): Unitigs deleted: 0\n",
            "\n",
            "CompactedDBG::construct(): Joining unitigs\n",
            "CompactedDBG::construct(): After join: 936536 unitigs\n",
            "CompactedDBG::construct(): Joined 55845 unitigs\n",
            "[build] building MPHF\n",
            "[build] creating equivalence classes ... \n",
            "[build] target de Bruijn graph has k-mer length 31 and minimizer length 23\n",
            "[build] target de Bruijn graph has 936536 contigs and contains 108619921 k-mers \n",
            "\n"
          ]
        }
      ],
      "source": [
        "!curl -O \\\n",
        "  ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
        "\n",
        "!./kallisto/build/src/kallisto index \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  Homo_sapiens.GRCh38.cdna.all.fa.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "z72tRWqruHyX"
      },
      "source": [
        "Непосредственно подсчёт экспрессии при помощи `kallisto`:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hh4YZMy9uLQW",
        "outputId": "d0b09be1-a262-43a0-99de-af4575e0f117"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: B1_SRR6410613_1.fastq.gz\n",
            "                             B1_SRR6410613_2.fastq.gz\n",
            "[progress] 49M reads processed (29.1% mapped)              done\n",
            "[quant] processed 49,995,049 reads, 14,568,421 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 137.0\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,088 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o results1 \\\n",
        "  B1_SRR6410613_1.fastq.gz \\\n",
        "  B1_SRR6410613_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nCI6OLJRLfnL",
        "outputId": "16240321-e6b2-4484-d584-21f86c259843"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: B2_SRR6410614_1.fastq.gz\n",
            "                             B2_SRR6410614_2.fastq.gz\n",
            "[progress] 49M reads processed (29.2% mapped)              done\n",
            "[quant] processed 49,996,267 reads, 14,605,289 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 142.7\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,273 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o results2 \\\n",
        "  B2_SRR6410614_1.fastq.gz \\\n",
        "  B2_SRR6410614_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "k5EWmgteBeQn",
        "outputId": "fa70d469-640a-4f59-bdec-9aa6688705ac"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: B3_SRR6410611_1.fastq.gz\n",
            "                             B3_SRR6410611_2.fastq.gz\n",
            "[progress] 49M reads processed (28.9% mapped)              done\n",
            "[quant] processed 49,997,942 reads, 14,448,747 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 138.3\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,576 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o results3 \\\n",
        "  B3_SRR6410611_1.fastq.gz \\\n",
        "  B3_SRR6410611_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "r1sN0dTDNFH_",
        "outputId": "7cb6a59f-c0e3-4ef0-ec4c-4485c92c83ad"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: B4_SRR6410612_1.fastq.gz\n",
            "                             B4_SRR6410612_2.fastq.gz\n",
            "[progress] 49M reads processed (28.5% mapped)              done\n",
            "[quant] processed 49,997,356 reads, 14,252,239 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 141.6\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,106 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o results3 \\\n",
        "  B4_SRR6410612_1.fastq.gz \\\n",
        "  B4_SRR6410612_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BY7MZ3PgEbJi",
        "outputId": "68c1766e-7e34-4e6e-d795-0e93ce8a35ef"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: B5_SRR6410615_1.fastq.gz\n",
            "                             B5_SRR6410615_2.fastq.gz\n",
            "[progress] 49M reads processed (84.2% mapped)              done\n",
            "[quant] processed 49,997,508 reads, 42,116,482 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 141.8\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,224 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o results5 \\\n",
        "  B5_SRR6410615_1.fastq.gz \\\n",
        "  B5_SRR6410615_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "76NTtZBAWMug",
        "outputId": "2df4e2d3-db59-4092-c242-19626250f187"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: A1_SRR6410605_1.fastq.gz\n",
            "                             A1_SRR6410605_2.fastq.gz\n",
            "[progress] 49M reads processed (54.9% mapped)              done\n",
            "[quant] processed 49,998,397 reads, 27,463,849 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 145.1\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,202 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o resultsA1 \\\n",
        "  A1_SRR6410605_1.fastq.gz \\\n",
        "  A1_SRR6410605_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UwJDdMGu3MND",
        "outputId": "58486fbd-b0b9-4479-f579-86676e0089bc"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: A2_SRR6410606_1.fastq.gz\n",
            "                             A2_SRR6410606_2.fastq.gz\n",
            "[progress] 49M reads processed (53.6% mapped)              done\n",
            "[quant] processed 49,996,562 reads, 26,808,329 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 144.6\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,251 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o resultsA2 \\\n",
        "  A2_SRR6410606_1.fastq.gz \\\n",
        "  A2_SRR6410606_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "W378hh9gMgMu",
        "outputId": "6dc1750e-252f-463d-fa1f-2729309274c0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: A3_SRR6410603_1.fastq.gz\n",
            "                             A3_SRR6410603_2.fastq.gz\n",
            "[progress] 49M reads processed (55.5% mapped)              done\n",
            "[quant] processed 50,005,922 reads, 27,736,221 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 152.1\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,174 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o resultsA3 \\\n",
        "  A3_SRR6410603_1.fastq.gz \\\n",
        "  A3_SRR6410603_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G_K5CyoVXmIz",
        "outputId": "e44d8b25-adf2-4ddf-f82a-ad706843f4e0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: A4_SRR6410604_1.fastq.gz\n",
            "                             A4_SRR6410604_2.fastq.gz\n",
            "[progress] 49M reads processed (54.6% mapped)              done\n",
            "[quant] processed 50,002,111 reads, 27,280,412 reads pseudoaligned\n",
            "[quant] estimated average fragment length: 145.2\n",
            "[   em] quantifying the abundances ... done\n",
            "[   em] the Expectation-Maximization algorithm ran for 1,616 rounds\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o resultsA4 \\\n",
        "  A4_SRR6410604_1.fastq.gz \\\n",
        "  A4_SRR6410604_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./kallisto/build/src/kallisto quant \\\n",
        "  -i Homo_sapiens.GRCh38.cdna.all.release-94_k31.idx \\\n",
        "  -o resultsA5 \\\n",
        "  A5_SRR6410609_1.fastq.gz \\\n",
        "  A5_SRR6410609_2.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "llPMpZCl-KdP",
        "outputId": "18b15b3a-53fb-41b3-8578-ad0a988c13eb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[quant] fragment length distribution will be estimated from the data\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 187,626\n",
            "[index] number of k-mers: 108,619,921\n",
            "[quant] running in paired-end mode\n",
            "[quant] will process pair 1: A5_SRR6410609_1.fastq.gz\n",
            "                             A5_SRR6410609_2.fastq.gz\n",
            "[progress] 39M reads processed (86.9% mapped)             "
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}