{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "from scipy.stats import fisher_exact\n",
        "from scipy import stats\n",
        "from statsmodels.stats.multitest import multipletests"
      ],
      "metadata": {
        "id": "2afQ4yZTCyXq"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "id": "sN1ZIDSxV20t"
      },
      "outputs": [],
      "source": [
        "expr1 = pd.read_csv(\"expr1.tsv\", sep=\"\\t\")\n",
        "anno1 = pd.read_csv(\"anno1.tsv\", sep=\"\\t\")\n",
        "genes1 = pd.read_csv(\"genes1.tsv\", sep=\"\\t\")\n",
        "expr2 = pd.read_csv(\"expr2.tsv\", sep=\"\\t\")\n",
        "anno2 = pd.read_csv(\"anno2.tsv\", sep=\"\\t\")\n",
        "genes2 = pd.read_csv(\"genes2.tsv\", sep=\"\\t\")\n",
        "expr3 = pd.read_csv(\"expr3.tsv\", sep=\"\\t\")\n",
        "anno3 = pd.read_csv(\"anno3.tsv\", sep=\"\\t\")\n",
        "genes3 = pd.read_csv(\"genes3.tsv\", sep=\"\\t\")\n",
        "expr4 = pd.read_csv(\"expr4.tsv\", sep=\"\\t\")\n",
        "anno4 = pd.read_csv(\"anno4.tsv\", sep=\"\\t\")\n",
        "genes4 = pd.read_csv(\"genes4.tsv\", sep=\"\\t\")\n",
        "expr5 = pd.read_csv(\"expr5.tsv\", sep=\"\\t\")\n",
        "anno5 = pd.read_csv(\"anno5.tsv\", sep=\"\\t\")\n",
        "genes5 = pd.read_csv(\"genes5.tsv\", sep=\"\\t\")\n",
        "expr6 = pd.read_csv(\"expr6.tsv\", sep=\"\\t\")\n",
        "anno6 = pd.read_csv(\"anno6.tsv\", sep=\"\\t\")\n",
        "genes6 = pd.read_csv(\"genes6.tsv\", sep=\"\\t\")\n",
        "expr7 = pd.read_csv(\"expr7.tsv\", sep=\"\\t\")\n",
        "anno7 = pd.read_csv(\"anno7.tsv\", sep=\"\\t\")\n",
        "genes7 = pd.read_csv(\"genes7.tsv\", sep=\"\\t\")\n",
        "expr8 = pd.read_csv(\"expr8.tsv\", sep=\"\\t\")\n",
        "anno8 = pd.read_csv(\"anno8.tsv\", sep=\"\\t\")\n",
        "genes8 = pd.read_csv(\"genes8.tsv\", sep=\"\\t\")\n",
        "expr9 = pd.read_csv(\"expr9.tsv\", sep=\"\\t\")\n",
        "anno9 = pd.read_csv(\"anno9.tsv\", sep=\"\\t\")\n",
        "genes9 = pd.read_csv(\"genes9.tsv\", sep=\"\\t\")\n",
        "expr10 = pd.read_csv(\"expr10.tsv\", sep=\"\\t\")\n",
        "anno10 = pd.read_csv(\"anno10.tsv\", sep=\"\\t\")\n",
        "genes10 = pd.read_csv(\"genes10.tsv\", sep=\"\\t\")\n",
        "expr11 = pd.read_csv(\"expr11.tsv\", sep=\"\\t\")\n",
        "anno11 = pd.read_csv(\"anno11.tsv\", sep=\"\\t\")\n",
        "genes11 = pd.read_csv(\"genes11.tsv\", sep=\"\\t\")\n",
        "expr12 = pd.read_csv(\"expr12.tsv\", sep=\"\\t\")\n",
        "anno12 = pd.read_csv(\"anno12.tsv\", sep=\"\\t\")\n",
        "genes12 = pd.read_csv(\"genes12.tsv\", sep=\"\\t\")\n",
        "expr13 = pd.read_csv(\"expr13.tsv\", sep=\"\\t\")\n",
        "anno13 = pd.read_csv(\"anno13.tsv\", sep=\"\\t\")\n",
        "genes13 = pd.read_csv(\"genes13.tsv\", sep=\"\\t\")"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "norm1 = expr1.iloc[:, 0:10]\n",
        "cancer1 = expr1.iloc[:, 10:]\n",
        "\n",
        "norm2 = expr2.iloc[:, 0:32]\n",
        "cancer2 = expr2.iloc[:, 32:]\n",
        "\n",
        "norm3 = expr3.iloc[:, 0:12]\n",
        "cancer3 = expr3.iloc[:, 12:]\n",
        "\n",
        "norm4 = expr4.iloc[:, 0:12]\n",
        "cancer4 = expr4.iloc[:, 12:]\n",
        "\n",
        "norm5 = expr5.iloc[:, 0:5]\n",
        "cancer5 = expr5.iloc[:, 5:]\n",
        "\n",
        "norm6 = expr6.iloc[:, 0:35]\n",
        "cancer6 = expr6.iloc[:, 35:]\n",
        "\n",
        "norm7 = expr7.iloc[:, 0:15]\n",
        "cancer7 = expr7.iloc[:, 15:]\n",
        "\n",
        "norm8 = expr8.iloc[:, 0:5]\n",
        "cancer8 = expr8.iloc[:, 5:]\n",
        "\n",
        "norm9 = expr9.iloc[:, 0:8]\n",
        "cancer9 = expr9.iloc[:, 8:]\n",
        "\n",
        "norm10 = expr10.iloc[:, 0:9]\n",
        "cancer10 = expr10.iloc[:, 9:]\n",
        "\n",
        "norm11 = expr11.iloc[:, 0:7]\n",
        "cancer11 = expr11.iloc[:, 7:]\n",
        "\n",
        "norm12 = expr12.iloc[:, 0:44]\n",
        "cancer12 = expr12.iloc[:, 44:]\n",
        "\n",
        "norm13 = expr13.iloc[:, 0:62]\n",
        "cancer13 = expr13.iloc[:, 62:]"
      ],
      "metadata": {
        "id": "GrSBzC4OZG5l"
      },
      "execution_count": 30,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "p1 = []\n",
        "for i in range(expr1.shape[0]):\n",
        "    p1.append(stats.mannwhitneyu(norm1.iloc[i, :],\n",
        "                                    cancer1.iloc[i, :]).pvalue)\n",
        "\n",
        "pv1 = multipletests(p1, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'1: Экспрессия {np.sum(pv1 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p2 = []\n",
        "for i in range(expr2.shape[0]):\n",
        "    p2.append(stats.mannwhitneyu(norm2.iloc[i, :],\n",
        "                                    cancer2.iloc[i, :]).pvalue)\n",
        "\n",
        "pv2 = multipletests(p2, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'2: Экспрессия {np.sum(pv2 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p3 = []\n",
        "for i in range(expr3.shape[0]):\n",
        "    p3.append(stats.mannwhitneyu(norm3.iloc[i, :],\n",
        "                                    cancer3.iloc[i, :]).pvalue)\n",
        "\n",
        "pv3 = multipletests(p3, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'3: Экспрессия {np.sum(pv3 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p4 = []\n",
        "for i in range(expr4.shape[0]):\n",
        "    p4.append(stats.mannwhitneyu(norm4.iloc[i, :],\n",
        "                                    cancer4.iloc[i, :]).pvalue)\n",
        "\n",
        "pv4 = multipletests(p4, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'4: Экспрессия {np.sum(pv4 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p5 = []\n",
        "for i in range(expr5.shape[0]):\n",
        "    p5.append(stats.mannwhitneyu(norm5.iloc[i, :],\n",
        "                                    cancer5.iloc[i, :]).pvalue)\n",
        "p5 = np.array(p5)\n",
        "#pv5 = multipletests(p5, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'5: Экспрессия {np.sum(p5 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p6 = []\n",
        "for i in range(expr6.shape[0]):\n",
        "    p6.append(stats.mannwhitneyu(norm6.iloc[i, :],\n",
        "                                    cancer6.iloc[i, :]).pvalue)\n",
        "\n",
        "pv6 = multipletests(p6, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'6: Экспрессия {np.sum(pv6 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p7 = []\n",
        "for i in range(expr7.shape[0]):\n",
        "    p7.append(stats.mannwhitneyu(norm7.iloc[i, :],\n",
        "                                    cancer7.iloc[i, :]).pvalue)\n",
        "\n",
        "pv7 = multipletests(p7, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'7: Экспрессия {np.sum(pv7 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p8 = []\n",
        "for i in range(expr8.shape[0]):\n",
        "    p8.append(stats.mannwhitneyu(norm8.iloc[i, :],\n",
        "                                    cancer8.iloc[i, :]).pvalue)\n",
        "p8 = np.array(p8)\n",
        "#pv8 = multipletests(p8, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'8: Экспрессия {np.sum(p8 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p9 = []\n",
        "for i in range(expr9.shape[0]):\n",
        "    p9.append(stats.mannwhitneyu(norm9.iloc[i, :],\n",
        "                                    cancer9.iloc[i, :]).pvalue)\n",
        "p9 = np.array(p9)\n",
        "#pv9 = multipletests(p9, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'9: Экспрессия {np.sum(p9 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p10 = []\n",
        "for i in range(expr10.shape[0]):\n",
        "    p10.append(stats.mannwhitneyu(norm10.iloc[i, :],\n",
        "                                    cancer10.iloc[i, :]).pvalue)\n",
        "\n",
        "pv10 = multipletests(p10, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'10: Экспрессия {np.sum(pv10 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p11 = []\n",
        "for i in range(expr11.shape[0]):\n",
        "    p11.append(stats.mannwhitneyu(norm11.iloc[i, :],\n",
        "                                    cancer11.iloc[i, :]).pvalue)\n",
        "p11 = np.array(p11)\n",
        "#pv11 = multipletests(p11, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'11: Экспрессия {np.sum(p11 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p12 = []\n",
        "for i in range(expr12.shape[0]):\n",
        "    p12.append(stats.mannwhitneyu(norm12.iloc[i, :],\n",
        "                                    cancer12.iloc[i, :]).pvalue)\n",
        "\n",
        "pv12 = multipletests(p12, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'12: Экспрессия {np.sum(pv12 < 0.05)} генов статистически достоверно различается между раком и нормой')\n",
        "\n",
        "p13 = []\n",
        "for i in range(expr13.shape[0]):\n",
        "    p13.append(stats.mannwhitneyu(norm13.iloc[i, :],\n",
        "                                    cancer13.iloc[i, :]).pvalue)\n",
        "\n",
        "pv13 = multipletests(p13, method=\"fdr_bh\")[1]\n",
        "\n",
        "print(f'13: Экспрессия {np.sum(pv13 < 0.05)} генов статистически достоверно различается между раком и нормой')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mIJOQCruX2gm",
        "outputId": "9de2f26b-f06c-4deb-de67-9dfcae2a49dc"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1: Экспрессия 2583 генов статистически достоверно различается между раком и нормой\n",
            "2: Экспрессия 10642 генов статистически достоверно различается между раком и нормой\n",
            "3: Экспрессия 10069 генов статистически достоверно различается между раком и нормой\n",
            "4: Экспрессия 7276 генов статистически достоверно различается между раком и нормой\n",
            "5: Экспрессия 2125 генов статистически достоверно различается между раком и нормой\n",
            "6: Экспрессия 13841 генов статистически достоверно различается между раком и нормой\n",
            "7: Экспрессия 2344 генов статистически достоверно различается между раком и нормой\n",
            "8: Экспрессия 1820 генов статистически достоверно различается между раком и нормой\n",
            "9: Экспрессия 863 генов статистически достоверно различается между раком и нормой\n",
            "10: Экспрессия 2734 генов статистически достоверно различается между раком и нормой\n",
            "11: Экспрессия 2792 генов статистически достоверно различается между раком и нормой\n",
            "12: Экспрессия 12707 генов статистически достоверно различается между раком и нормой\n",
            "13: Экспрессия 13979 генов статистически достоверно различается между раком и нормой\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "genes_under_study1 = genes1[\"rownames(eset)\"][pv1 < 0.05].apply(str)\n",
        "background_genes1 = genes1[\"rownames(eset)\"][pv1 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study2 = genes2[\"rownames(eset)\"][pv2 < 0.05].apply(str)\n",
        "background_genes2 = genes2[\"rownames(eset)\"][pv2 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study3 = genes3[\"rownames(eset)\"][pv3 < 0.05].apply(str)\n",
        "background_genes3 = genes3[\"rownames(eset)\"][pv3 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study4 = genes4[\"rownames(eset)\"][pv4 < 0.05].apply(str)\n",
        "background_genes4 = genes4[\"rownames(eset)\"][pv4 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study5 = genes5[\"rownames(eset)\"][p5 < 0.05].apply(str)\n",
        "background_genes5 = genes5[\"rownames(eset)\"][p5 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study6 = genes6[\"rownames(eset)\"][pv6 < 0.05].apply(str)\n",
        "background_genes6 = genes6[\"rownames(eset)\"][pv6 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study7 = genes7[\"rownames(eset)\"][pv7 < 0.05].apply(str)\n",
        "background_genes7 = genes7[\"rownames(eset)\"][pv7 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study8 = genes8[\"rownames(eset)\"][p8 < 0.05].apply(str)\n",
        "background_genes8 = genes8[\"rownames(eset)\"][p8 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study9 = genes9[\"rownames(eset)\"][p9 < 0.05].apply(str)\n",
        "background_genes9 = genes9[\"rownames(eset)\"][p9 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study10 = genes10[\"rownames(eset)\"][pv10 < 0.05].apply(str)\n",
        "background_genes10 = genes10[\"rownames(eset)\"][pv10 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study11 = genes11[\"rownames(eset)\"][p11 < 0.05].apply(str)\n",
        "background_genes11 = genes11[\"rownames(eset)\"][p11 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study12 = genes12[\"rownames(eset)\"][pv12 < 0.05].apply(str)\n",
        "background_genes12 = genes12[\"rownames(eset)\"][pv12 >= 0.05].apply(str)\n",
        "\n",
        "genes_under_study13 = genes13[\"rownames(eset)\"][pv13 < 0.05].apply(str)\n",
        "background_genes13 = genes13[\"rownames(eset)\"][pv13 >= 0.05].apply(str)"
      ],
      "metadata": {
        "id": "SwN4SNpkbFz8"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes1:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study1:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "id": "cQMwaq-fjVkb",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cc7339a4-1523-454b-f8f9-ff7b1a5b8d9a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_ARACHIDONIC_ACID_METABOLISM',\n",
              "       'KEGG_PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
              "       'KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC',\n",
              "       'KEGG_CITRATE_CYCLE_TCA_CYCLE', 'KEGG_FATTY_ACID_METABOLISM',\n",
              "       'KEGG_OXIDATIVE_PHOSPHORYLATION',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_LYSINE_DEGRADATION', 'KEGG_TYROSINE_METABOLISM',\n",
              "       'KEGG_TRYPTOPHAN_METABOLISM', 'KEGG_BETA_ALANINE_METABOLISM',\n",
              "       'KEGG_GLUTATHIONE_METABOLISM', 'KEGG_PROPANOATE_METABOLISM',\n",
              "       'KEGG_BUTANOATE_METABOLISM',\n",
              "       'KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM',\n",
              "       'KEGG_TERPENOID_BACKBONE_BIOSYNTHESIS',\n",
              "       'KEGG_LIMONENE_AND_PINENE_DEGRADATION',\n",
              "       'KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450',\n",
              "       'KEGG_DRUG_METABOLISM_CYTOCHROME_P450',\n",
              "       'KEGG_BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS',\n",
              "       'KEGG_PPAR_SIGNALING_PATHWAY', 'KEGG_MAPK_SIGNALING_PATHWAY',\n",
              "       'KEGG_PEROXISOME', 'KEGG_APOPTOSIS',\n",
              "       'KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION',\n",
              "       'KEGG_WNT_SIGNALING_PATHWAY', 'KEGG_AXON_GUIDANCE',\n",
              "       'KEGG_FOCAL_ADHESION', 'KEGG_ECM_RECEPTOR_INTERACTION',\n",
              "       'KEGG_TIGHT_JUNCTION', 'KEGG_GAP_JUNCTION',\n",
              "       'KEGG_ALZHEIMERS_DISEASE', 'KEGG_PARKINSONS_DISEASE',\n",
              "       'KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS',\n",
              "       'KEGG_HUNTINGTONS_DISEASE', 'KEGG_PATHWAYS_IN_CANCER'],\n",
              "      dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 54
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes2:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study2:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9km-rJ5HhxLq",
        "outputId": "2e4b1bf0-68bd-4b80-ac41-e7ddf0f9161b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_N_GLYCAN_BIOSYNTHESIS', 'KEGG_GLYCEROLIPID_METABOLISM',\n",
              "       'KEGG_AMINOACYL_TRNA_BIOSYNTHESIS',\n",
              "       'KEGG_NON_HOMOLOGOUS_END_JOINING', 'KEGG_LYSOSOME',\n",
              "       'KEGG_CIRCADIAN_RHYTHM_MAMMAL', 'KEGG_PENTOSE_PHOSPHATE_PATHWAY',\n",
              "       'KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
              "       'KEGG_FATTY_ACID_METABOLISM', 'KEGG_PURINE_METABOLISM',\n",
              "       'KEGG_PYRIMIDINE_METABOLISM',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_ARGININE_AND_PROLINE_METABOLISM', 'KEGG_SULFUR_METABOLISM',\n",
              "       'KEGG_RIBOSOME', 'KEGG_RNA_DEGRADATION', 'KEGG_RNA_POLYMERASE',\n",
              "       'KEGG_DNA_REPLICATION', 'KEGG_SPLICEOSOME', 'KEGG_PROTEASOME',\n",
              "       'KEGG_BASE_EXCISION_REPAIR', 'KEGG_NUCLEOTIDE_EXCISION_REPAIR',\n",
              "       'KEGG_MISMATCH_REPAIR', 'KEGG_HOMOLOGOUS_RECOMBINATION',\n",
              "       'KEGG_MAPK_SIGNALING_PATHWAY', 'KEGG_CHEMOKINE_SIGNALING_PATHWAY',\n",
              "       'KEGG_CELL_CYCLE', 'KEGG_OOCYTE_MEIOSIS',\n",
              "       'KEGG_P53_SIGNALING_PATHWAY',\n",
              "       'KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS', 'KEGG_APOPTOSIS',\n",
              "       'KEGG_WNT_SIGNALING_PATHWAY', 'KEGG_AXON_GUIDANCE',\n",
              "       'KEGG_CELL_ADHESION_MOLECULES_CAMS', 'KEGG_GAP_JUNCTION',\n",
              "       'KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
              "       'KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS',\n",
              "       'KEGG_REGULATION_OF_ACTIN_CYTOSKELETON',\n",
              "       'KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
              "       'KEGG_PRION_DISEASES', 'KEGG_VIBRIO_CHOLERAE_INFECTION',\n",
              "       'KEGG_PATHWAYS_IN_CANCER', 'KEGG_COLORECTAL_CANCER',\n",
              "       'KEGG_SMALL_CELL_LUNG_CANCER', 'KEGG_VIRAL_MYOCARDITIS'],\n",
              "      dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 55
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes3:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study3:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pck2uLzximYX",
        "outputId": "4927e6f0-d3e3-4692-a362-061acb35f524"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_N_GLYCAN_BIOSYNTHESIS', 'KEGG_GLYCEROLIPID_METABOLISM',\n",
              "       'KEGG_GLYCEROPHOSPHOLIPID_METABOLISM',\n",
              "       'KEGG_GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES',\n",
              "       'KEGG_PANTOTHENATE_AND_COA_BIOSYNTHESIS',\n",
              "       'KEGG_OXIDATIVE_PHOSPHORYLATION', 'KEGG_PURINE_METABOLISM',\n",
              "       'KEGG_PYRIMIDINE_METABOLISM',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_INOSITOL_PHOSPHATE_METABOLISM', 'KEGG_PROPANOATE_METABOLISM',\n",
              "       'KEGG_RIBOSOME', 'KEGG_RNA_POLYMERASE', 'KEGG_DNA_REPLICATION',\n",
              "       'KEGG_SPLICEOSOME', 'KEGG_PROTEASOME',\n",
              "       'KEGG_MAPK_SIGNALING_PATHWAY', 'KEGG_ERBB_SIGNALING_PATHWAY',\n",
              "       'KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
              "       'KEGG_CHEMOKINE_SIGNALING_PATHWAY',\n",
              "       'KEGG_PHOSPHATIDYLINOSITOL_SIGNALING_SYSTEM', 'KEGG_CELL_CYCLE',\n",
              "       'KEGG_OOCYTE_MEIOSIS', 'KEGG_P53_SIGNALING_PATHWAY',\n",
              "       'KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS', 'KEGG_PEROXISOME',\n",
              "       'KEGG_MTOR_SIGNALING_PATHWAY', 'KEGG_APOPTOSIS',\n",
              "       'KEGG_WNT_SIGNALING_PATHWAY', 'KEGG_TGF_BETA_SIGNALING_PATHWAY',\n",
              "       'KEGG_AXON_GUIDANCE', 'KEGG_FOCAL_ADHESION', 'KEGG_GAP_JUNCTION',\n",
              "       'KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
              "       'KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
              "       'KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS',\n",
              "       'KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
              "       'KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION',\n",
              "       'KEGG_ALZHEIMERS_DISEASE', 'KEGG_PARKINSONS_DISEASE',\n",
              "       'KEGG_VIBRIO_CHOLERAE_INFECTION',\n",
              "       'KEGG_EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION',\n",
              "       'KEGG_PATHWAYS_IN_CANCER', 'KEGG_COLORECTAL_CANCER',\n",
              "       'KEGG_RENAL_CELL_CARCINOMA', 'KEGG_PANCREATIC_CANCER',\n",
              "       'KEGG_ENDOMETRIAL_CANCER', 'KEGG_GLIOMA', 'KEGG_PROSTATE_CANCER',\n",
              "       'KEGG_BLADDER_CANCER', 'KEGG_SMALL_CELL_LUNG_CANCER',\n",
              "       'KEGG_NON_SMALL_CELL_LUNG_CANCER'], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes4:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study4:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vp26UAX3pB5I",
        "outputId": "f81f6a91-8cac-41ea-e8a2-3206e33f2815"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_LYSOSOME', 'KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION',\n",
              "       'KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION',\n",
              "       'KEGG_PRIMARY_IMMUNODEFICIENCY', 'KEGG_GLYCOLYSIS_GLUCONEOGENESIS',\n",
              "       'KEGG_CITRATE_CYCLE_TCA_CYCLE',\n",
              "       'KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM',\n",
              "       'KEGG_FATTY_ACID_METABOLISM', 'KEGG_OXIDATIVE_PHOSPHORYLATION',\n",
              "       'KEGG_PURINE_METABOLISM', 'KEGG_PYRIMIDINE_METABOLISM',\n",
              "       'KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM',\n",
              "       'KEGG_GLYCINE_SERINE_AND_THREONINE_METABOLISM',\n",
              "       'KEGG_CYSTEINE_AND_METHIONINE_METABOLISM',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_LYSINE_DEGRADATION', 'KEGG_ARGININE_AND_PROLINE_METABOLISM',\n",
              "       'KEGG_BETA_ALANINE_METABOLISM', 'KEGG_GLUTATHIONE_METABOLISM',\n",
              "       'KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM',\n",
              "       'KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE',\n",
              "       'KEGG_INOSITOL_PHOSPHATE_METABOLISM', 'KEGG_PYRUVATE_METABOLISM',\n",
              "       'KEGG_PROPANOATE_METABOLISM', 'KEGG_BUTANOATE_METABOLISM',\n",
              "       'KEGG_ONE_CARBON_POOL_BY_FOLATE', 'KEGG_RNA_DEGRADATION',\n",
              "       'KEGG_DNA_REPLICATION', 'KEGG_PROTEASOME', 'KEGG_PROTEIN_EXPORT',\n",
              "       'KEGG_PPAR_SIGNALING_PATHWAY', 'KEGG_BASE_EXCISION_REPAIR',\n",
              "       'KEGG_NUCLEOTIDE_EXCISION_REPAIR', 'KEGG_HOMOLOGOUS_RECOMBINATION',\n",
              "       'KEGG_ERBB_SIGNALING_PATHWAY',\n",
              "       'KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION', 'KEGG_CELL_CYCLE',\n",
              "       'KEGG_OOCYTE_MEIOSIS', 'KEGG_P53_SIGNALING_PATHWAY',\n",
              "       'KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS', 'KEGG_ENDOCYTOSIS',\n",
              "       'KEGG_CELL_ADHESION_MOLECULES_CAMS', 'KEGG_TIGHT_JUNCTION',\n",
              "       'KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY',\n",
              "       'KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY',\n",
              "       'KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS',\n",
              "       'KEGG_LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION',\n",
              "       'KEGG_INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION',\n",
              "       'KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
              "       'KEGG_TYPE_I_DIABETES_MELLITUS',\n",
              "       'KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION',\n",
              "       'KEGG_ALZHEIMERS_DISEASE', 'KEGG_PARKINSONS_DISEASE',\n",
              "       'KEGG_VIBRIO_CHOLERAE_INFECTION',\n",
              "       'KEGG_EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION',\n",
              "       'KEGG_LEISHMANIA_INFECTION', 'KEGG_PANCREATIC_CANCER',\n",
              "       'KEGG_BLADDER_CANCER', 'KEGG_NON_SMALL_CELL_LUNG_CANCER',\n",
              "       'KEGG_ALLOGRAFT_REJECTION', 'KEGG_GRAFT_VERSUS_HOST_DISEASE',\n",
              "       'KEGG_VIRAL_MYOCARDITIS'], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes5:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study5:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nTXOQjQ9pWoT",
        "outputId": "f0f1e4f5-d6f2-482b-b664-7159cdbc0545"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_CARDIAC_MUSCLE_CONTRACTION',\n",
              "       'KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION',\n",
              "       'KEGG_GLYCOLYSIS_GLUCONEOGENESIS',\n",
              "       'KEGG_OXIDATIVE_PHOSPHORYLATION',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_ANTIGEN_PROCESSING_AND_PRESENTATION',\n",
              "       'KEGG_TYPE_I_DIABETES_MELLITUS', 'KEGG_ALZHEIMERS_DISEASE',\n",
              "       'KEGG_PARKINSONS_DISEASE', 'KEGG_HUNTINGTONS_DISEASE',\n",
              "       'KEGG_LEISHMANIA_INFECTION', 'KEGG_AUTOIMMUNE_THYROID_DISEASE',\n",
              "       'KEGG_ALLOGRAFT_REJECTION', 'KEGG_GRAFT_VERSUS_HOST_DISEASE',\n",
              "       'KEGG_VIRAL_MYOCARDITIS'], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes6:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study6:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XBlUAgDFp2JF",
        "outputId": "b3355953-0d6c-48ee-d7ee-d24c76785454"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC',\n",
              "       'KEGG_DILATED_CARDIOMYOPATHY', 'KEGG_CALCIUM_SIGNALING_PATHWAY',\n",
              "       'KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION', 'KEGG_FOCAL_ADHESION',\n",
              "       'KEGG_LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION',\n",
              "       'KEGG_REGULATION_OF_ACTIN_CYTOSKELETON', 'KEGG_PATHWAYS_IN_CANCER'],\n",
              "      dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes7:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study7:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TAwQ3L1bqEj7",
        "outputId": "3b43c204-e2f9-4a4d-e1a7-e1350aed65db"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_O_GLYCAN_BIOSYNTHESIS',\n",
              "       'KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION',\n",
              "       'KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC',\n",
              "       'KEGG_GLYCOLYSIS_GLUCONEOGENESIS', 'KEGG_PURINE_METABOLISM',\n",
              "       'KEGG_PROPANOATE_METABOLISM', 'KEGG_ERBB_SIGNALING_PATHWAY',\n",
              "       'KEGG_CELL_CYCLE', 'KEGG_AXON_GUIDANCE', 'KEGG_FOCAL_ADHESION',\n",
              "       'KEGG_TIGHT_JUNCTION', 'KEGG_REGULATION_OF_ACTIN_CYTOSKELETON',\n",
              "       'KEGG_PATHWAYS_IN_CANCER', 'KEGG_SMALL_CELL_LUNG_CANCER'],\n",
              "      dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes8:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study8:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gapxP0zDqSJz",
        "outputId": "62bf1ae8-8069-47dd-8892-20f07b4242ef"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array([], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes9:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study9:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5IstclYcqdTU",
        "outputId": "2d341ed2-1dba-4c73-e41c-391a5edd4e56"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array([], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes10:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study10:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xrlg23Bmqjda",
        "outputId": "d1275ec0-5d42-460c-cdf3-ee2a71183055"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_GLYCEROPHOSPHOLIPID_METABOLISM',\n",
              "       'KEGG_SYSTEMIC_LUPUS_ERYTHEMATOSUS',\n",
              "       'KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION',\n",
              "       'KEGG_PYRUVATE_METABOLISM', 'KEGG_PROPANOATE_METABOLISM',\n",
              "       'KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
              "       'KEGG_AXON_GUIDANCE', 'KEGG_CELL_ADHESION_MOLECULES_CAMS',\n",
              "       'KEGG_INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION',\n",
              "       'KEGG_LEISHMANIA_INFECTION', 'KEGG_ASTHMA',\n",
              "       'KEGG_VIRAL_MYOCARDITIS'], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes11:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study11:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kV6KwghGqrsJ",
        "outputId": "0772533f-5561-46c1-e5d7-6d0ab61a844b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_GLYCEROLIPID_METABOLISM',\n",
              "       'KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION',\n",
              "       'KEGG_P53_SIGNALING_PATHWAY', 'KEGG_FOCAL_ADHESION',\n",
              "       'KEGG_ECM_RECEPTOR_INTERACTION',\n",
              "       'KEGG_COMPLEMENT_AND_COAGULATION_CASCADES',\n",
              "       'KEGG_PATHWAYS_IN_CANCER', 'KEGG_SMALL_CELL_LUNG_CANCER'],\n",
              "      dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes12:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study12:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bZoPh_Whq0MQ",
        "outputId": "ba4b333a-1662-44c0-c729-f5ae0798e1c1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array(['KEGG_N_GLYCAN_BIOSYNTHESIS', 'KEGG_AMINOACYL_TRNA_BIOSYNTHESIS',\n",
              "       'KEGG_SYSTEMIC_LUPUS_ERYTHEMATOSUS', 'KEGG_PURINE_METABOLISM',\n",
              "       'KEGG_PYRIMIDINE_METABOLISM', 'KEGG_GLUTATHIONE_METABOLISM',\n",
              "       'KEGG_RNA_DEGRADATION', 'KEGG_RNA_POLYMERASE', 'KEGG_SPLICEOSOME',\n",
              "       'KEGG_PROTEASOME', 'KEGG_CELL_CYCLE', 'KEGG_OOCYTE_MEIOSIS',\n",
              "       'KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS', 'KEGG_ENDOCYTOSIS',\n",
              "       'KEGG_AXON_GUIDANCE', 'KEGG_VEGF_SIGNALING_PATHWAY',\n",
              "       'KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS',\n",
              "       'KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION',\n",
              "       'KEGG_PATHWAYS_IN_CANCER'], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kegg_pw = {}\n",
        "for line in open(\"c2.cp.kegg.v7.5.symbols.gmt.txt\"):\n",
        "    parts = line.strip().split(\"\\t\")\n",
        "    name, desc, genes_lst = parts[0], parts[1], parts[2:]\n",
        "    kegg_pw[name] = set(genes_lst)\n",
        "\n",
        "pw_pv = []\n",
        "pw = []\n",
        "for pathw, pw_genes in kegg_pw.items():\n",
        "    background_genes_in = 0\n",
        "    background_genes_out = 0\n",
        "    genes_under_study_in = 0\n",
        "    genes_under_study_out = 0\n",
        "    for gene in background_genes13:\n",
        "        if gene in pw_genes:\n",
        "            background_genes_in += 1\n",
        "        else:\n",
        "            background_genes_out += 1\n",
        "    for gene in genes_under_study13:\n",
        "        if gene in pw_genes:\n",
        "            genes_under_study_in += 1\n",
        "        else:\n",
        "            genes_under_study_out += 1\n",
        "    table = np.array([[genes_under_study_in, background_genes_in], [genes_under_study_out, background_genes_out]])\n",
        "    pw_pv.append(fisher_exact(table, \"greater\")[1])\n",
        "    pw.append(pathw)\n",
        "\n",
        "np.array(pw)[multipletests(pw_pv, method=\"fdr_bh\")[1] < 0.05]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zYFdZx8Oq-jH",
        "outputId": "032ddf91-2ff2-4517-8d95-bd2b7bbbcbf9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "array([], dtype='<U63')"
            ]
          },
          "metadata": {},
          "execution_count": 18
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "genes_under_study1 = genes1[\"rownames(eset)\"][expr1.iloc[:,-1]>5].apply(str)\n",
        "myfile1 = genes_under_study1.tolist()\n",
        "with open('cancer1.txt', 'w') as f:\n",
        "     print(myfile1, file=f)\n",
        "\n",
        "genes_under_study2 = genes2[\"rownames(eset)\"][expr2.iloc[:,-1]>5].apply(str)\n",
        "myfile2 = genes_under_study2.tolist()\n",
        "with open('cancer2.txt', 'w') as f:\n",
        "     print(myfile2, file=f)\n",
        "\n",
        "genes_under_study3 = genes3[\"rownames(eset)\"][expr3.iloc[:,-1]>5].apply(str)\n",
        "myfile3 = genes_under_study3.tolist()\n",
        "with open('cancer3.txt', 'w') as f:\n",
        "     print(myfile3, file=f)\n",
        "\n",
        "genes_under_study4 = genes4[\"rownames(eset)\"][expr4.iloc[:,-1]>5].apply(str)\n",
        "myfile4 = genes_under_study4.tolist()\n",
        "with open('cancer4.txt', 'w') as f:\n",
        "     print(myfile4, file=f)\n",
        "\n",
        "genes_under_study5 = genes5[\"rownames(eset)\"][expr5.iloc[:,-1]>5].apply(str)\n",
        "myfile5 = genes_under_study5.tolist()\n",
        "with open('cancer5.txt', 'w') as f:\n",
        "     print(myfile5, file=f)\n",
        "\n",
        "genes_under_study6 = genes6[\"rownames(eset)\"][expr6.iloc[:,-1]>5].apply(str)\n",
        "myfile6 = genes_under_study6.tolist()\n",
        "with open('cancer6.txt', 'w') as f:\n",
        "     print(myfile6, file=f)\n",
        "\n",
        "genes_under_study7 = genes7[\"rownames(eset)\"][expr7.iloc[:,-1]>5].apply(str)\n",
        "myfile7 = genes_under_study7.tolist()\n",
        "with open('cancer7.txt', 'w') as f:\n",
        "     print(myfile7, file=f)\n",
        "\n",
        "genes_under_study8 = genes8[\"rownames(eset)\"][expr8.iloc[:,-1]>5].apply(str)\n",
        "myfile8 = genes_under_study8.tolist()\n",
        "with open('cancer8.txt', 'w') as f:\n",
        "     print(myfile8, file=f)\n",
        "\n",
        "genes_under_study9 = genes9[\"rownames(eset)\"][expr9.iloc[:,-1]>5].apply(str)\n",
        "myfile9 = genes_under_study9.tolist()\n",
        "with open('cancer9.txt', 'w') as f:\n",
        "     print(myfile9, file=f)\n",
        "\n",
        "genes_under_study10 = genes10[\"rownames(eset)\"][expr10.iloc[:,-1]>5].apply(str)\n",
        "myfile10 = genes_under_study10.tolist()\n",
        "with open('cancer10.txt', 'w') as f:\n",
        "     print(myfile10, file=f)\n",
        "\n",
        "genes_under_study11 = genes11[\"rownames(eset)\"][expr11.iloc[:,-1]>5].apply(str)\n",
        "myfile11 = genes_under_study11.tolist()\n",
        "with open('cancer11.txt', 'w') as f:\n",
        "     print(myfile11, file=f)\n",
        "\n",
        "genes_under_study12 = genes12[\"rownames(eset)\"][expr12.iloc[:,-1]>5].apply(str)\n",
        "myfile12 = genes_under_study12.tolist()\n",
        "with open('cancer12.txt', 'w') as f:\n",
        "     print(myfile12, file=f)\n",
        "\n",
        "genes_under_study13 = genes13[\"rownames(eset)\"][expr13.iloc[:,-1]>5].apply(str)\n",
        "myfile13 = genes_under_study13.tolist()\n",
        "with open('cancer13.txt', 'w') as f:\n",
        "     print(myfile13, file=f)"
      ],
      "metadata": {
        "id": "f2uSDbpVefjO"
      },
      "execution_count": 22,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "genes_under_study1 = genes1[\"rownames(eset)\"][expr1.iloc[:,0]>5].apply(str)\n",
        "myfile1 = genes_under_study1.tolist()\n",
        "with open('norm1_1.txt', 'w') as f:\n",
        "     print(myfile1, file=f)\n",
        "\n",
        "genes_under_study2 = genes2[\"rownames(eset)\"][expr2.iloc[:,0]>5].apply(str)\n",
        "myfile2 = genes_under_study2.tolist()\n",
        "with open('norm2_1.txt', 'w') as f:\n",
        "     print(myfile2, file=f)\n",
        "\n",
        "genes_under_study3 = genes3[\"rownames(eset)\"][expr3.iloc[:,0]>5].apply(str)\n",
        "myfile3 = genes_under_study3.tolist()\n",
        "with open('norm3_1.txt', 'w') as f:\n",
        "     print(myfile3, file=f)\n",
        "\n",
        "genes_under_study4 = genes4[\"rownames(eset)\"][expr4.iloc[:,0]>5].apply(str)\n",
        "myfile4 = genes_under_study4.tolist()\n",
        "with open('norm4_1.txt', 'w') as f:\n",
        "     print(myfile4, file=f)\n",
        "\n",
        "genes_under_study5 = genes5[\"rownames(eset)\"][expr5.iloc[:,0]>5].apply(str)\n",
        "myfile5 = genes_under_study5.tolist()\n",
        "with open('norm5_1.txt', 'w') as f:\n",
        "     print(myfile5, file=f)\n",
        "\n",
        "genes_under_study6 = genes6[\"rownames(eset)\"][expr6.iloc[:,0]>5].apply(str)\n",
        "myfile6 = genes_under_study6.tolist()\n",
        "with open('norm6_1.txt', 'w') as f:\n",
        "     print(myfile6, file=f)\n",
        "\n",
        "genes_under_study7 = genes7[\"rownames(eset)\"][expr7.iloc[:,0]>5].apply(str)\n",
        "myfile7 = genes_under_study7.tolist()\n",
        "with open('norm7_1.txt', 'w') as f:\n",
        "     print(myfile7, file=f)\n",
        "\n",
        "genes_under_study8 = genes8[\"rownames(eset)\"][expr8.iloc[:,0]>5].apply(str)\n",
        "myfile8 = genes_under_study8.tolist()\n",
        "with open('norm8_1.txt', 'w') as f:\n",
        "     print(myfile8, file=f)\n",
        "\n",
        "genes_under_study9 = genes9[\"rownames(eset)\"][expr9.iloc[:,0]>5].apply(str)\n",
        "myfile9 = genes_under_study9.tolist()\n",
        "with open('norm9_1.txt', 'w') as f:\n",
        "     print(myfile9, file=f)\n",
        "\n",
        "genes_under_study10 = genes10[\"rownames(eset)\"][expr10.iloc[:,0]>5].apply(str)\n",
        "myfile10 = genes_under_study10.tolist()\n",
        "with open('norm10_1.txt', 'w') as f:\n",
        "     print(myfile10, file=f)\n",
        "\n",
        "genes_under_study11 = genes11[\"rownames(eset)\"][expr11.iloc[:,0]>5].apply(str)\n",
        "myfile11 = genes_under_study11.tolist()\n",
        "with open('norm11_1.txt', 'w') as f:\n",
        "     print(myfile11, file=f)\n",
        "\n",
        "genes_under_study12 = genes12[\"rownames(eset)\"][expr12.iloc[:,0]>5].apply(str)\n",
        "myfile12 = genes_under_study12.tolist()\n",
        "with open('norm12_1.txt', 'w') as f:\n",
        "     print(myfile12, file=f)\n",
        "\n",
        "genes_under_study13 = genes13[\"rownames(eset)\"][expr13.iloc[:,0]>5].apply(str)\n",
        "myfile13 = genes_under_study13.tolist()\n",
        "with open('norm13_1.txt', 'w') as f:\n",
        "     print(myfile13, file=f)"
      ],
      "metadata": {
        "id": "52AYiz9JfC82"
      },
      "execution_count": 23,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "norm1 = norm1>5\n",
        "cancer1 = cancer1>5\n",
        "\n",
        "norm2 = norm2>5\n",
        "cancer2 = cancer2>5\n",
        "\n",
        "norm3 = norm3>5\n",
        "cancer3 = cancer3>5\n",
        "\n",
        "norm4 = norm4>5\n",
        "cancer4 = cancer4>5\n",
        "\n",
        "norm5 = norm5>5\n",
        "cancer5 = cancer5>5\n",
        "\n",
        "norm6 = norm6>5\n",
        "cancer6 = cancer6>5\n",
        "\n",
        "norm7 = norm7>5\n",
        "cancer7 = cancer7>5\n",
        "\n",
        "norm8 = norm8>5\n",
        "cancer8 = cancer8>5\n",
        "\n",
        "norm9 = norm9>5\n",
        "cancer9 = cancer9>5\n",
        "\n",
        "norm10 = norm10>5\n",
        "cancer10 = cancer10>5\n",
        "\n",
        "norm11 = norm11>5\n",
        "cancer11 = cancer11>5\n",
        "\n",
        "norm12 = norm12>5\n",
        "cancer12 = cancer12>5\n",
        "\n",
        "norm13 = norm13>5\n",
        "cancer13 = cancer13>5"
      ],
      "metadata": {
        "id": "rkgQ69wqGkez"
      },
      "execution_count": 31,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "norm1['new_col'] = norm1.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer1['new_col'] = cancer1.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm2['new_col'] = norm2.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer2['new_col'] = cancer2.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm3['new_col'] = norm3.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer3['new_col'] = cancer3.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm4['new_col'] = norm4.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer4['new_col'] = cancer4.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm5['new_col'] = norm5.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer5['new_col'] = cancer5.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm6['new_col'] = norm6.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer6['new_col'] = cancer6.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm7['new_col'] = norm7.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer7['new_col'] = cancer7.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm8['new_col'] = norm8.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer8['new_col'] = cancer8.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm9['new_col'] = norm9.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer9['new_col'] = cancer9.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm10['new_col'] = norm10.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer10['new_col'] = cancer10.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm11['new_col'] = norm11.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer11['new_col'] = cancer11.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm12['new_col'] = norm12.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer12['new_col'] = cancer12.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "\n",
        "norm13['new_col'] = norm13.apply(lambda row: True if True in row.values else False, axis=1)\n",
        "cancer13['new_col'] = cancer13.apply(lambda row: True if True in row.values else False, axis=1)"
      ],
      "metadata": {
        "id": "yZT0lC0JHgTH"
      },
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "genes_under_study1 = genes1[\"rownames(eset)\"][cancer1['new_col']].apply(str)\n",
        "myfile1 = genes_under_study1.tolist()\n",
        "with open('cancer1.txt', 'w') as f:\n",
        "     print(myfile1, file=f)\n",
        "\n",
        "genes_under_study2 = genes2[\"rownames(eset)\"][cancer2['new_col']].apply(str)\n",
        "myfile2 = genes_under_study2.tolist()\n",
        "with open('cancer2.txt', 'w') as f:\n",
        "     print(myfile2, file=f)\n",
        "\n",
        "genes_under_study3 = genes3[\"rownames(eset)\"][cancer3['new_col']].apply(str)\n",
        "myfile3 = genes_under_study3.tolist()\n",
        "with open('cancer3.txt', 'w') as f:\n",
        "     print(myfile3, file=f)\n",
        "\n",
        "genes_under_study4 = genes4[\"rownames(eset)\"][cancer4['new_col']].apply(str)\n",
        "myfile4 = genes_under_study4.tolist()\n",
        "with open('cancer4.txt', 'w') as f:\n",
        "     print(myfile4, file=f)\n",
        "\n",
        "genes_under_study5 = genes5[\"rownames(eset)\"][cancer5['new_col']].apply(str)\n",
        "myfile5 = genes_under_study5.tolist()\n",
        "with open('cancer5.txt', 'w') as f:\n",
        "     print(myfile5, file=f)\n",
        "\n",
        "genes_under_study6 = genes6[\"rownames(eset)\"][cancer6['new_col']].apply(str)\n",
        "myfile6 = genes_under_study6.tolist()\n",
        "with open('cancer6.txt', 'w') as f:\n",
        "     print(myfile6, file=f)\n",
        "\n",
        "genes_under_study7 = genes7[\"rownames(eset)\"][cancer7['new_col']].apply(str)\n",
        "myfile7 = genes_under_study7.tolist()\n",
        "with open('cancer7.txt', 'w') as f:\n",
        "     print(myfile7, file=f)\n",
        "\n",
        "genes_under_study8 = genes8[\"rownames(eset)\"][cancer8['new_col']].apply(str)\n",
        "myfile8 = genes_under_study8.tolist()\n",
        "with open('cancer8.txt', 'w') as f:\n",
        "     print(myfile8, file=f)\n",
        "\n",
        "genes_under_study9 = genes9[\"rownames(eset)\"][cancer9['new_col']].apply(str)\n",
        "myfile9 = genes_under_study9.tolist()\n",
        "with open('cancer9.txt', 'w') as f:\n",
        "     print(myfile9, file=f)\n",
        "\n",
        "genes_under_study10 = genes10[\"rownames(eset)\"][cancer10['new_col']].apply(str)\n",
        "myfile10 = genes_under_study10.tolist()\n",
        "with open('cancer10.txt', 'w') as f:\n",
        "     print(myfile10, file=f)\n",
        "\n",
        "genes_under_study11 = genes11[\"rownames(eset)\"][cancer11['new_col']].apply(str)\n",
        "myfile11 = genes_under_study11.tolist()\n",
        "with open('cancer11.txt', 'w') as f:\n",
        "     print(myfile11, file=f)\n",
        "\n",
        "genes_under_study12 = genes12[\"rownames(eset)\"][cancer12['new_col']].apply(str)\n",
        "myfile12 = genes_under_study12.tolist()\n",
        "with open('cancer12.txt', 'w') as f:\n",
        "     print(myfile12, file=f)\n",
        "\n",
        "genes_under_study13 = genes13[\"rownames(eset)\"][cancer13['new_col']].apply(str)\n",
        "myfile13 = genes_under_study13.tolist()\n",
        "with open('cancer13.txt', 'w') as f:\n",
        "     print(myfile13, file=f)"
      ],
      "metadata": {
        "id": "PHgKPLrO1yOH"
      },
      "execution_count": 27,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "genes_under_study1 = genes1[\"rownames(eset)\"][norm1['new_col']].apply(str)\n",
        "myfile1 = genes_under_study1.tolist()\n",
        "with open('norm1_1.txt', 'w') as f:\n",
        "     print(myfile1, file=f)\n",
        "\n",
        "genes_under_study2 = genes2[\"rownames(eset)\"][norm2['new_col']].apply(str)\n",
        "myfile2 = genes_under_study2.tolist()\n",
        "with open('norm2_1.txt', 'w') as f:\n",
        "     print(myfile2, file=f)\n",
        "\n",
        "genes_under_study3 = genes3[\"rownames(eset)\"][norm3['new_col']].apply(str)\n",
        "myfile3 = genes_under_study3.tolist()\n",
        "with open('norm3_1.txt', 'w') as f:\n",
        "     print(myfile3, file=f)\n",
        "\n",
        "genes_under_study4 = genes4[\"rownames(eset)\"][norm4['new_col']].apply(str)\n",
        "myfile4 = genes_under_study4.tolist()\n",
        "with open('norm4_1.txt', 'w') as f:\n",
        "     print(myfile4, file=f)\n",
        "\n",
        "genes_under_study5 = genes5[\"rownames(eset)\"][norm5['new_col']].apply(str)\n",
        "myfile5 = genes_under_study5.tolist()\n",
        "with open('norm5_1.txt', 'w') as f:\n",
        "     print(myfile5, file=f)\n",
        "\n",
        "genes_under_study6 = genes6[\"rownames(eset)\"][norm6['new_col']].apply(str)\n",
        "myfile6 = genes_under_study6.tolist()\n",
        "with open('norm6_1.txt', 'w') as f:\n",
        "     print(myfile6, file=f)\n",
        "\n",
        "genes_under_study7 = genes7[\"rownames(eset)\"][norm7['new_col']].apply(str)\n",
        "myfile7 = genes_under_study7.tolist()\n",
        "with open('norm7_1.txt', 'w') as f:\n",
        "     print(myfile7, file=f)\n",
        "\n",
        "genes_under_study8 = genes8[\"rownames(eset)\"][norm8['new_col']].apply(str)\n",
        "myfile8 = genes_under_study8.tolist()\n",
        "with open('norm8_1.txt', 'w') as f:\n",
        "     print(myfile8, file=f)\n",
        "\n",
        "genes_under_study9 = genes9[\"rownames(eset)\"][norm9['new_col']].apply(str)\n",
        "myfile9 = genes_under_study9.tolist()\n",
        "with open('norm9_1.txt', 'w') as f:\n",
        "     print(myfile9, file=f)\n",
        "\n",
        "genes_under_study10 = genes10[\"rownames(eset)\"][norm10['new_col']].apply(str)\n",
        "myfile10 = genes_under_study10.tolist()\n",
        "with open('norm10_1.txt', 'w') as f:\n",
        "     print(myfile10, file=f)\n",
        "\n",
        "genes_under_study11 = genes11[\"rownames(eset)\"][norm11['new_col']].apply(str)\n",
        "myfile11 = genes_under_study11.tolist()\n",
        "with open('norm11_1.txt', 'w') as f:\n",
        "     print(myfile11, file=f)\n",
        "\n",
        "genes_under_study12 = genes12[\"rownames(eset)\"][norm12['new_col']].apply(str)\n",
        "myfile12 = genes_under_study12.tolist()\n",
        "with open('norm12_1.txt', 'w') as f:\n",
        "     print(myfile12, file=f)\n",
        "\n",
        "genes_under_study13 = genes13[\"rownames(eset)\"][norm13['new_col']].apply(str)\n",
        "myfile13 = genes_under_study13.tolist()\n",
        "with open('norm13_1.txt', 'w') as f:\n",
        "     print(myfile13, file=f)"
      ],
      "metadata": {
        "id": "iSOnU2gTwIvm"
      },
      "execution_count": 28,
      "outputs": []
    }
  ]
}